### PR TITLE
Create user on `CompleteProfile` form submit

### PR DESCRIPTION
**Pull request by henrikwinston**

### CODE REVIEW

File: app/(auth)/user/complete-profile/components/index.tsx [hunk 1]
```diff
@@ -8,6 +10,7 @@ import * as z from 'zod';
 
 export * from './step-0-pack';
 export * from './step-1-pack';
+export * from './description-pack';
```
Comment (imsumoregn) [inline app/(auth)/user/complete-profile/components/index.tsx:13]: Keeping having to do this is not very ideal - what would be the solution if we take forward compatibility in to account?

File: app/(auth)/user/complete-profile/components/step-2-pack.tsx [hunk 2]
```diff
@@ -32,7 +32,7 @@ export const Step2InputPack = () => (
       }}
     />
     <FormField
-      name="github"
+      name="githubUrl"
```
Comment (imsumoregn) [inline app/(auth)/user/complete-profile/components/step-2-pack.tsx:35]: Usually in bigger teams, there would be a centralised solution for these kind of things from day zero.

File: app/(auth)/user/complete-profile/page.tsx [hunk 3]
```diff
@@ -26,6 +27,7 @@ import { CheckCircleIcon } from '@heroicons/react/outline';
 import { ArrowLeftIcon } from '@radix-ui/react-icons';
 import { zodResolver } from '@hookform/resolvers/zod';
 import { useForm } from 'react-hook-form';
+import { api } from '~/lib/trpc/client';
```
Comment (imsumoregn) [inline app/(auth)/user/complete-profile/page.tsx]: Nit: Import order.

File: lib/trpc/server.ts [hunk 4]
```diff
@@ -2,10 +2,10 @@ import { httpBatchLink } from '@trpc/client';
...
 import { getBaseUrl } from './shared';
 
-export const rscApi = appRouter.createCaller({
-  links: [
-    httpBatchLink({
-      url: getBaseUrl() + '/api/trpc',
-    }),
-  ],
-});
+// export const rscApi = appRouter.createCaller({
```
Comment (imsumoregn) [inline lib/trpc/server.ts:5]: What happened here?

File: package.json [hunk 5]
```diff
@@ -27,47 +27,47 @@
...
     "class-variance-authority": "^0.7.0",
     "clsx": "^2.0.0",
     "cmdk": "^0.2.0",
     "drizzle-orm": "^0.26.5",
     "formik": "^2.4.5",
-    "framer-motion": "^10.16.12",
+    "framer-motion": "^10.16.16",
     "jwt-decode": "^3.1.2",
-    "next": "13.5.4",
+    "next": "14.0.4",
```
Comment (imsumoregn) [inline package.json:44]: For big version jump, let's have it in a separate PR next time, for the sake of reverting.

File: server/api/routers/user.ts [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+  major: z.string().min(2).max(50),
+  title: z.string().min(2).max(50),
+  workplace: z.string().min(2).max(50),
+  location: z.string().min(2).max(50),
+  githubUrl: z.string().min(2).max(50),
+  linkedinUrl: z.string().min(2).max(50),
+  fields: z.array(z.string()),
+  offers: z.array(z.string()),
+  bookingUrl: z.string().min(2).max(50),
+});
```
Comment (imsumoregn) [inline server/api/routers/user.ts:30]: If you prefer to have the schema here, then the `formSchema` on the client could be inferred from the router output.
https://trpc.io/docs/client/vanilla/infer-types

File: server/db/schema.prisma [hunk 7]
```diff
@@ -21,29 +21,30 @@ enum Gender {
 
 model Field {
   fieldId     Int    @id @default(autoincrement()) @map("field_id")
-  name        String @db.VarChar(25)
+  name        String @unique @db.VarChar(25)
```
Comment (imsumoregn) [inline server/db/schema.prisma:24]: Not bad.

**Review by imsumoregn — CHANGES_REQUESTED** [review]

File: server/api/routers/user.ts (hunk 6 above)
Comment (henrikwinston) [reply server/api/routers/user.ts:30]: Issue created

File: package.json (hunk 5 above)
Comment (henrikwinston) [reply package.json:44]: yb

File: lib/trpc/server.ts (hunk 4 above)
Comment (henrikwinston) [reply lib/trpc/server.ts:5]: The previous code worked because there was no context, but after adding db and auth, I can't put the context in here

Comment (henrikwinston) [reply lib/trpc/server.ts:5]: So I decided to delay the fixing until we actually need to use this

File: app/(auth)/user/complete-profile/components/index.tsx (hunk 1 above)
Comment (henrikwinston) [reply app/(auth)/user/complete-profile/components/index.tsx:13]: Maybe to store those form parts in a map or something, created an issue

File: app/(auth)/user/complete-profile/components/step-2-pack.tsx (hunk 2 above)
Comment (henrikwinston) [reply app/(auth)/user/complete-profile/components/step-2-pack.tsx:35]: You mean having the keys extracted from the schema?

File: app/(auth)/user/complete-profile/components/index.tsx (hunk 1 above)
Comment (imsumoregn) [reply app/(auth)/user/complete-profile/components/index.tsx:13]: Next milestone, then.

File: app/(auth)/user/complete-profile/components/step-2-pack.tsx (hunk 2 above)
Comment (imsumoregn) [reply app/(auth)/user/complete-profile/components/step-2-pack.tsx:35]: Could be - you can make decisions, just please do have reasons for them.

**Review by imsumoregn — APPROVED** [review]
Skipping failed deployment since we are not using `@ant-design/icons` anymore.

**Pull request merged**